File: CollaborationFront/WebContent/Friend/friendservice.js
/**
 * FRIEND SERVICE
 */
app.factory('FriendService',function($http){

	var friendService={}
	var BASE_URL="http://localhost/CollaborationMiddle"
    friendService.listOfSuggestedUsers=function()
    {
		return $http.get(BASE_URL+"/getsuggestedusers")
    }

	friendService.sendFriendRequest=function(toId)
    {
		return $http.get(BASE_URL+"/friendRequest/"+toId)
    }
	friendService.pendingRequests=function()
    {
		return $http.get(BASE_URL+"/pendingRequests")
    }
	friendService.listOfFriends=function()
	{
		return $http.get(BASE_URL+"/listOfFriends")
	}
	friendService.updatePendingRequest=function()
	{
		return $http.put(BASE_URL+"/updatePendingRequests")
	}
    return friendService;
})

File: CollaborationBack/src/main/java/com/collab/CollaborationBack/Service/FriendService.java
package com.collab.CollaborationBack.Service;

import java.util.List;

import com.collab.CollaborationBack.model.Friend;
import com.collab.CollaborationBack.model.User;

public interface FriendService {
	 List<User> listofsuggestedusers(String userName);
	 void createfriend(Friend friend);
	 List<Friend> pendingrequest(String toId);
	 List<String> listOfFriends(String userName);
	void updatePendingRequests(Friend friend);
	
	
}
